src/crimson/osd/osd.cc

```diff
@@ -0,0 +1,386 @@
+#include "osd.h"
+
+#include "messages/MOSDBeacon.h"
+#include "messages/MOSDBoot.h"
+#include "messages/MOSDMap.h"
+#include "crimson/net/Connection.h"
+#include "crimson/net/Messenger.h"
+
+namespace {
+  seastar::logger& logger() {
+    return ceph::get_logger(ceph_subsys_osd);
+  }
+
+  template<typename Message, typename... Args>
+  Ref<Message> make_message(Args&&... args)
+  {
+    return {new Message{std::forward<Args>(args)...}, false};
+  }
+}
+
+using ceph::common::local_conf;
+
+OSD::OSD(int id,
+         ceph::net::Messenger& cluster_msgr,
+         ceph::net::Messenger& client_msgr)
+  : whoami{id},
+    cluster_msgr{cluster_msgr},
+    client_msgr{client_msgr},
+    monc{client_msgr}
+{
+  osdmaps[0] = seastar::make_lw_shared<OSDMap>();
+  dispatchers.push_front(this);
+  dispatchers.push_front(&monc);
+  beacon_timer.set_callback([this] {
+    send_beacon();
+  });
+}
+
```

Comment:
here, if mkfs is called before reactor start, we can't return future, if it is called after reactor start, CyanStore::write_meta will cause block.  so  I think we can decided when to execute it and how to implement write_meta first.

Reply:
please note, i don't see "mkfs is called before reactor start" as a need here. again,  "block" is not a show blocker, as long as it is not noticed by the reactor's cpu stall detector, and it is not sitting in the critical path.

Reply:
i agree, ideally, we should not have blocked calls in seastar thread, but i think we can always address the blocking calls if we have extra bandwidth.